subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/EnvironmentConfigurator.kt

```diff
@@ -0,0 +1,43 @@
+package com.avito.instrumentation.internal
+
+import com.avito.instrumentation.InstrumentationTestsTask
+import com.avito.instrumentation.configuration.ExecutionEnvironment
+import com.avito.instrumentation.configuration.KubernetesViaContext
+import com.avito.instrumentation.configuration.KubernetesViaCredentials
+import com.avito.instrumentation.configuration.LocalAdb
+import com.avito.utils.gradle.KubernetesCredentials
+
+internal class EnvironmentConfigurator(
+    private val environment: ExecutionEnvironment
+) : InstrumentationTaskConfigurator {
+
+    override fun configure(task: InstrumentationTestsTask) {
+        when (environment) {
+            is KubernetesViaCredentials ->
+                task.kubernetesCredentials.set(
+                    KubernetesCredentials.Service(
```

Comment:
Looks like we want to reuse those models

Reply:
I'm not sure that we need to reuse models for gradle public API,
if it will be covered by proposed dsl abstraction: https://github.com/avito-tech/avito-android/pull/1265#discussion_r748098479 then yes